Dockerfile.common

```diff
@@ -0,0 +1,117 @@
+ && apt-get install -qy curl                               \
+ && curl -sL https://deb.nodesource.com/setup_8.x | bash - \
+ && apt-get update                                         \
+ && apt-get install -qy                                    \
+        build-essential                                    \
+        git                                                \
+        libcairo-dev                                       \
+        libcurl4-openssl-dev                               \
+        libffi-dev                                         \
+        libjpeg-dev                                        \
+        libjpeg8-dev                                       \
+        libldap2-dev                                       \
+        libpython-dev                                      \
+        libsasl2-dev                                       \
+        libssl-dev                                         \
+        libxml2-dev                                        \
+        libxslt1-dev                                       \
+        nodejs                                             \
+        openssl                                            \
+        python                                             \
+        python-pip                                         \
+        wget                                               \
+        zlib1g-dev                                         \
+ && apt-get build-dep -qy                                  \
+        gdal                                               \
+ && pip install -U pip                                     \
+ && pip install setuptools                                 \
+ && apt-get clean                                          \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN echo 'NODE VERSION' \
+ && echo '------------' \
+ && node --version      \
+ && echo                \
+ && echo 'NPM VERSION'  \
+ && echo '-----------'  \
+ && npm --version
+
+ARG MAKE_PARALLELISM
+
+RUN mkdir /tmp/gdal                                                     \
+ && curl -s -o - http://download.osgeo.org/gdal/2.2.3/gdal-2.2.3.tar.gz \
+ |  tar -xzf - -C /tmp/gdal --strip-components 1                        \
+ && cd /tmp/gdal                                                        \
+ && ./configure --prefix=/usr                                           \
+ && make $MAKE_PARALLELISM                                              \
+ && make install                                                        \
+ && cd /                                                                \
+ && rm -rf /tmp/gdal
+
+ENV CPLUS_INCLUDE_PATH=/usr/include/gdal
+ENV C_INCLUDE_PATH=/usr/include/gdal
+ENV C_FORCE_ROOT=1
+
+RUN pip install GDAL==$(gdal-config --version)
+
+RUN mkdir -p /girder-components/plugins
+
+ADD girder /girder
+
+RUN pip install -e /girder[plugins]                            \
+ && pip install -r /girder/requirements-dev.txt                \
```

Comment:
The latest girder installs its testing package via a relative path (https://github.com/girder/girder/blob/master/requirements-dev.txt#L1).  Pip currently resolves lines like these relative to the working directory rather than the path of the requirements file.  This means you will have to `cd /girder` before running this command.